source/Calamari.Tests/Fixtures/PackageRetention/JournalEntryFixture.cs

```diff
@@ -128,12 +139,30 @@ public void WhenPackageIsRegisteredAndDeregistered_ThenUsageIsStillRecorded()
             var thePackage = new PackageIdentity("Package", "1.0");
             var deploymentOne = new ServerTaskId("Deployment-1");
 
-            var journal = new Journal(new InMemoryJournalRepositoryFactory(), variables, Substitute.For<ILog>());
+            var journal = GetPackageJournal();
             journal.RegisterPackageUse(thePackage, deploymentOne);
             journal.DeregisterPackageUse(thePackage, deploymentOne);
 
             Assert.AreEqual(1, journal.GetUsage(thePackage).Count());
         }
+        
+        [Test]
+        public void ApplyRetention_RemovesPackageFile_AndUsage()
+        {
+            var thePackage = new PackageIdentity("Package", "1.0");
+            var deploymentOne = new ServerTaskId("Deployment-1");
+    
+            var journal = GetPackageJournal();
+            journal.RegisterPackageUse(thePackage, deploymentOne);
+
+            journal.ApplyRetention(PackageDirectory);
+
+            // TODO: Fix these assertions
+            // How do we ensure the given package will be marked for removal?
+            // How do we simulate needing more disk space
```

Comment:
You can use a real `IFreeSpaceChecker`, with a mock `IFileSystem` passed to it - and make the mock `fileSystem.GetDiskFreeSpace` call return the space you want it to.
Let me know if you want me to walk you through it.